source/common/upstream/upstream_impl.cc

```diff
@@ -1089,6 +1089,9 @@ ClusterInfoImpl::ClusterInfoImpl(
                         ? std::make_unique<envoy::config::cluster::v3::Cluster::CustomClusterType>(
                               config.cluster_type())
                         : nullptr),
+      http_filter_config_provider_manager_(
```

Comment:
I did try this, but the problem is that since the config provider manager depends on ``ClusterManager``, this kind of change would require adding a function to get a reference to config provider manager in the ``ClusterManager`` and therefore create a circular dependency.

https://github.com/envoyproxy/envoy/blob/14e139e6a310530f4290adebd5ebd03aae9b5098/envoy/filter/config_provider_manager.h#L54

Reply:
I see, can you write this down as TODO? We can follow-up with a refactor to break the cycle later.

Reply:
done, added in upstream_impl.h where the field is defined